NSubstitute.DbConnection/NSubstitute.DbConnection.Tests/QueryParameterMatchingTests.cs

```diff
@@ -249,4 +250,86 @@ public void ShouldFailWhenSetUpWithNoParameters()
         });
         result.Should().Throw<NotSupportedException>().WithMessage(NoMatchingQueryErrorMessage+$": '{commandText}'");
     }
+
+    [Test]
+    public void ShouldIncludeOutputParametersWhenMatching()
+    {
+        var mockConnection = Substitute.For<IDbConnection>().SetupCommands();
+        var commandText = "spMySproc";
+        var record = new KeyValueRecord(1, "abc");
+        mockConnection.SetupQuery(commandText)
+            .WithParameters(new Dictionary<string, object> { { "id", 1 } })
+            .WithOutputParameters(("output", 7))
```

Comment:
Just checking that the value passed in here is the mock output value for the parameter, as opposed to an input value?

Is it ever valid for an output parameter to have a value passed in I wonder? Could that passed-in value ever be relevant to the execution of a query, i.e. could its value be read by the query before it runs, before then being set overwriting the existing value when the query finishes running?